### DISCUSSION TITLE
Remote Development - Visual Studio Code

### DISCUSSION THREAD

**Discussion by skywalkerisnull** [gravitational/teleport#5741] [Q&A] [answered]
Hello,

A number of our developers are using [VS Code Remote Development](https://code.visualstudio.com/docs/remote/remote-overview) has anyone configured Teleport so that they may continue to use VS Code (or any other SSH based remote development tools) to remote into and continue to develop? And how have you set that up?

**Comment by klizhentas** [comment, marked as answer]
May be not exactly 100% answer to your question, but you can use OpenSSH clients to work with teleport:

https://github.com/gravitational/teleport/discussions/5530

In this case, any tool that supports standard SSH should work, including VSCode. I recall @stevenGravy and @webvictim were discussing VSCode with Teleport too

**Reply by webvictim** [reply]
Yes, I use VS Code with Teleport using the setup detailed in #5530. The only thing I remember that I had to change was to set `remote.SSH.useLocalServer` to `false` in extension settings - this may not be the case any more but I haven't seen any reason to change it as things work fine.

**Reply by skywalkerisnull** [reply]
@klizhentas and @webvictim thank you for the link, I will look into those and test it out

**Comment by travelton** [comment]
In addition to the above setting that @webvictim suggests, your local SSH configuration needs to be configured properly. If your ProxyCommand differs, try the below instead. We've observed other ProxyCommand syntax to cause VSCode to error with `Failed to parse remote port from server` (a generic unrelated error). 

```
Host example-node-1
  User root
  Port 3022
  ProxyCommand ssh -p 3023 %r@teleport-proxy.example.com -s proxy:%h:%p
```

Note: Add `-A` to your ProxyCommand if you have Teleport Proxy Recording mode enabled or if you wish to forward your local agent to the target remote node. Keep in mind Teleport automatically loads your Teleport SSH keys into your local agent if it's running.

**Comment by benarent** [comment]
We've since created a video to highlight this feature https://www.youtube.com/watch?v=uL1bnhUelP8
